lib/mediator-service/index.js

```diff
@@ -0,0 +1,260 @@
+var CONSTANTS = require('../constants');
+var shortid = require('shortid');
+var q = require('q');
+var MediatorTopicUtility = require('fh-wfm-mediator/lib/topics');
+
+/**
+ *
+ * Getting Promises for done and error topics.
+ * This will resolve or reject the reuturned promise depending on the topic published.
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {Promise}
+ */
+function getTopicPromises(doneTopicPromise, errorTopicPromise) {
+  var deferred = q.defer();
+
+  doneTopicPromise.then(function(createdWorkorder) {
+    deferred.resolve(createdWorkorder);
+  });
+
+  errorTopicPromise.then(function(error) {
+    deferred.reject(error);
+  });
+
+  return deferred.promise;
+}
+
+
+
+/**
+ *
+ * A mediator service that will publish and subscribe to topics to be able to render data.
+ *
+ * @param {Mediator} mediator
+ * @param {object}   config
+ * @constructor
+ */
+function WorkorderMediatorService(mediator, config) {
+  this.mediator = mediator;
+  this.config = config || {};
+
+  this.workordersTopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.WORKORDER_ENTITY_NAME);
+  this.usersTopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.USERS_ENTITY_NAME);
+  this.workflowsTopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.WORKFLOWS_ENTITY_NAME);
+  this.resultsTopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.RESULTS_ENTITY_NAME);
```

Comment:
Ok, wrapping the mediator callbacks into service classes is really, really cool. +1